app/routes/_gcn.circulars.edit.$circularId/CircularEditForm.tsx

```diff
@@ -0,0 +1,353 @@
+/*!
+ * Copyright © 2023 United States Government as represented by the Administrator
+ * of the National Aeronautics and Space Administration. No copyright is claimed
+ * in the United States under Title 17, U.S. Code. All Other Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { Form, Link, useNavigation } from '@remix-run/react'
+import {
+  Button,
+  ButtonGroup,
+  Icon,
+  InputGroup,
+  InputPrefix,
+  Table,
+  TextInput,
+  Textarea,
+} from '@trussworks/react-uswds'
+import classnames from 'classnames'
+import type { ReactNode } from 'react'
+import { useContext, useState } from 'react'
+import { dedent } from 'ts-dedent'
+
+import { AstroDataContext } from '../_gcn.circulars.$circularId.($version)/AstroDataContext'
+import {
+  MarkdownBody,
+  PlainTextBody,
+} from '../_gcn.circulars.$circularId.($version)/Body'
+import type { Circular } from '../_gcn.circulars/circulars.lib'
+import { bodyIsValid, subjectIsValid } from '../_gcn.circulars/circulars.lib'
+import { CircularsKeywords } from '~/components/CircularsKeywords'
+import Spinner from '~/components/Spinner'
+
+function SyntaxExample({
+  label,
+  children,
+  href,
+  ...props
+}: {
+  label: ReactNode
+  children: string
+} & Pick<JSX.IntrinsicElements['a'], 'href' | 'rel'>) {
+  const { target, rel } = useContext(AstroDataContext)
+  return (
+    <tr>
+      <td>
+        {href ? (
+          <a href={href} {...props} target={target} rel={rel}>
+            {label}
+          </a>
+        ) : (
+          label
+        )}
+      </td>
+      <td>
+        <code>{children}</code>
+      </td>
+      <td>
+        <MarkdownBody>{children}</MarkdownBody>
+      </td>
+    </tr>
+  )
+}
+
+function SyntaxReference() {
+  return (
+    <>
+      <h3>GCN Circulars cross-reference syntax</h3>
+      <Table compact bordered>
+        <thead>
+          <tr>
+            <th>Service</th>
+            <th>Example</th>
+            <th>Output</th>
+          </tr>
+        </thead>
+        <tbody>
+          <SyntaxExample label="GCN Circulars" href="/circulars">
+            GCN 34653, 34660, and 34677
+          </SyntaxExample>
+          <SyntaxExample label="DOI" href="https://www.doi.org" rel="external">
+            doi:10.1103/PhysRevLett.116.061102
+          </SyntaxExample>
+          <SyntaxExample label="arXiv" href="https://arxiv.org" rel="external">
+            arXiv:1602.03837
+          </SyntaxExample>
+          <SyntaxExample
+            label="TNS"
+            href="https://www.wis-tns.org"
+            rel="external"
+          >
+            AT2017gfo
+          </SyntaxExample>
+          <SyntaxExample label="Web address">
+            https://www.swift.psu.edu
+          </SyntaxExample>
+        </tbody>
+      </Table>
+    </>
+  )
+}
+
+export default function CircularEditForm({
```

Comment:
I'm trying to get out of the habit of using default exports, except in route modules.

```suggestion
export function CircularEditForm({
```

Reply:
Gotcha, I'll include this in #1784